chat-plugins/trivia.js

```diff
@@ -0,0 +1,1225 @@
+	// the behaviour of this method can be entirely arbitrary otherwise.
+	answerQuestion() {}
+
+	// Verifies whether or not an answer is correct. In longer answers, small
+	// typos can be made and still have the answer be considered correct.
+	verifyAnswer(tarAnswer) {
+		return this.curAnswers.some(answer => (
+			(answer === tarAnswer) || (answer.length > 5 && Tools.levenshtein(tarAnswer, answer) < 3)
+		));
+	}
+
+	// This is a noop here since it'd defined properly by mode decorators later
+	// on. This calculates the points a correct responder earns, which is
+	// typically between 1-5.
+	calculatePoints() {}
+
+	// This is a noop here since it's defined properly by mode decorators later
+	// on. This is obligated to update the game phase, but it can be entirely
+	// arbitrary otherwise.
+	tallyAnswers() {}
+
+	// Ends the game after a player's score has exceeded the score cap.
+	// FIXME: this class and trivia database logic don't belong in bed with
+	// each other! Abstract all that away from this method as soon as possible.
+	win(winner, buffer) {
+		buffer += Tools.escapeHTML(winner.name) + ' won the game with a final score of <strong>' + winner.points + '</strong>, ' +
+			'and their leaderboard score has increased by <strong>' + this.prize + '</strong> points!';
+		this.broadcast('The answering period has ended!', buffer);
+
+		let leaderboard = triviaData.leaderboard;
+		for (let i in this.players) {
+			let player = this.players[i];
+			if (!player.points) continue;
+
+			if (leaderboard[i]) {
+				let rank = leaderboard[i];
+				rank[1] += player.points;
+				rank[2] += player.correctAnswers;
+			} else {
+				leaderboard[i] = [0, player.points, player.correctAnswers];
+			}
+		}
+
+		if (winner) leaderboard[winner.userid][0] += this.prize;
+
+		let leaders = Object.keys(leaderboard);
+		let ladder = triviaData.ladder = [];
+		for (let i = 0; i < 3; i++) {
+			leaders.sort((a, b) => leaderboard[b][i] - leaderboard[a][i]);
+
+			let max = Infinity;
+			let rank = 0;
+			let rankIdx = i + 3;
+			for (let j = 0; j < leaders.length; j++) {
+				let leader = leaders[j];
+				let score = leaderboard[leader][i];
+				if (max !== score) {
+					if (!i && rank < 15) {
+						if (ladder[rank]) {
+							ladder[rank].push(leader);
+						} else {
+							ladder[rank] = [leader];
+						}
+					}
+
+					rank++;
+					max = score;
+				}
+				leaderboard[leader][rankIdx] = rank;
+			}
+		}
+
+		writeTriviaData();
+		this.destroy();
+	}
+
+	// Forcibly ends a trivia game.
+	end(user) {
+		this.broadcast('The trivia game was forcibly ended by ' + Tools.escapeHTML(user.name) + '.');
+		this.destroy();
+	}
+}
+
+// Subclassing is slow! On top of that, subclassing to define the same three
+// methods every time does not make much sense. Instead, we'll define the
+// methods used to answer questions, calculate points, and reward correct
+// answerers by using a decorator for each mode.
```

Comment:
>Subclassing is slow!

As if that matters in non-performance critical code

>subclassing to define the same three methods every time does not make much sense

That's the entire point of interfaces - to have subclasses define the same common method/property names

You should definitely be subclassing here